Build error on clang 13.0.1

When building with clang13.0.1 I get errors related to narrowing conversions see below.

Compiler info:
```console
Homebrew clang version 13.0.1
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /opt/homebrew/opt/llvm/bin
```

```console
  Running setup.py install for GeodisTK ... error
  error: subprocess-exited-with-error
  
  × Running setup.py install for GeodisTK did not run successfully.
  │ exit code: 1
  ╰─> [79 lines of output]
      running install
      /Users/jacobmerson/geodis/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
        warnings.warn(
      running build
      running build_ext
      building 'GeodisTK' extension
      creating build
      creating build/temp.macosx-12-arm64-cpython-39
      creating build/temp.macosx-12-arm64-cpython-39/cpp
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -I/Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include -I./cpp -I/Users/jacobmerson/geodis/include -I/opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c ./cpp/geodesic_distance.cpp -o build/temp.macosx-12-arm64-cpython-39/./cpp/geodesic_distance.o
      In file included from ./cpp/geodesic_distance.cpp:3:
      In file included from /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/arrayobject.h:5:
      In file included from /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/ndarrayobject.h:12:
      In file included from /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/ndarraytypes.h:1948:
      /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API, disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
      #warning "Using deprecated NumPy API, disable it with " \
       ^
      ./cpp/geodesic_distance.cpp:28:16: warning: unused variable 'shape_seed' [-Wunused-variable]
          npy_intp * shape_seed = PyArray_DIMS(arr_Seed);
                     ^
      ./cpp/geodesic_distance.cpp:76:16: warning: unused variable 'shape_seed' [-Wunused-variable]
          npy_intp * shape_seed = PyArray_DIMS(arr_Seed);
                     ^
      ./cpp/geodesic_distance.cpp:126:16: warning: unused variable 'shape_seed' [-Wunused-variable]
          npy_intp * shape_seed = PyArray_DIMS(arr_Seed);
                     ^
      ./cpp/geodesic_distance.cpp:185:16: warning: unused variable 'shape_seed' [-Wunused-variable]
          npy_intp * shape_seed = PyArray_DIMS(arr_Seed);
                     ^
      In file included from ./cpp/geodesic_distance.cpp:3:
      In file included from /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/arrayobject.h:5:
      In file included from /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/ndarrayobject.h:21:
      /Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include/numpy/__multiarray_api.h:1477:1: warning: unused function '_import_array' [-Wunused-function]
      _import_array(void)
      ^
      ./cpp/geodesic_distance.cpp:222:20: warning: unused variable 'Methods' [-Wunused-variable]
      static PyMethodDef Methods[] = {
                         ^
      7 warnings generated.
      clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk -I/Users/jacobmerson/geodis/lib/python3.9/site-packages/numpy/core/include -I./cpp -I/Users/jacobmerson/geodis/include -I/opt/homebrew/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/include/python3.9 -c ./cpp/geodesic_distance_2d.cpp -o build/temp.macosx-12-arm64-cpython-39/./cpp/geodesic_distance_2d.o
      ./cpp/geodesic_distance_2d.cpp:12:22: warning: comparison of integers of different signs: 'int' and 'std::vector<float>::size_type' (aka 'unsigned long') [-Wsign-compare]
          for(int d = 0; d < p1.size(); d++)
                         ~ ^ ~~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:23:35: warning: comparison of integers of different signs: 'int' and 'std::vector<Point2D>::size_type' (aka 'unsigned long') [-Wsign-compare]
          for(int i = start_position; i < list->size(); i++)
                                      ~ ^ ~~~~~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:37:22: warning: comparison of integers of different signs: 'int' and 'std::vector<Point2D>::size_type' (aka 'unsigned long') [-Wsign-compare]
          for(int i = 0; i < list->size(); i++)
                         ~ ^ ~~~~~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:175:33: error: non-constant-expression cannot be narrowed from type 'double' to 'float' in initializer list [-Wc++11-narrowing]
              float local_dis_f[4] = {sqrt(2.0), 1.0, sqrt(2.0), 1.0};
                                      ^~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:175:33: note: insert an explicit cast to silence this issue
              float local_dis_f[4] = {sqrt(2.0), 1.0, sqrt(2.0), 1.0};
                                      ^~~~~~~~~
                                      static_cast<float>( )
      ./cpp/geodesic_distance_2d.cpp:175:49: error: non-constant-expression cannot be narrowed from type 'double' to 'float' in initializer list [-Wc++11-narrowing]
              float local_dis_f[4] = {sqrt(2.0), 1.0, sqrt(2.0), 1.0};
                                                      ^~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:175:49: note: insert an explicit cast to silence this issue
              float local_dis_f[4] = {sqrt(2.0), 1.0, sqrt(2.0), 1.0};
                                                      ^~~~~~~~~
                                                      static_cast<float>( )
      ./cpp/geodesic_distance_2d.cpp:203:38: error: non-constant-expression cannot be narrowed from type 'double' to 'float' in initializer list [-Wc++11-narrowing]
              float local_dis_b[4] = {1.0, sqrt(2.0), 1.0, sqrt(2.0)};
                                           ^~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:203:38: note: insert an explicit cast to silence this issue
              float local_dis_b[4] = {1.0, sqrt(2.0), 1.0, sqrt(2.0)};
                                           ^~~~~~~~~
                                           static_cast<float>( )
      ./cpp/geodesic_distance_2d.cpp:203:54: error: non-constant-expression cannot be narrowed from type 'double' to 'float' in initializer list [-Wc++11-narrowing]
              float local_dis_b[4] = {1.0, sqrt(2.0), 1.0, sqrt(2.0)};
                                                           ^~~~~~~~~
      ./cpp/geodesic_distance_2d.cpp:203:54: note: insert an explicit cast to silence this issue
              float local_dis_b[4] = {1.0, sqrt(2.0), 1.0, sqrt(2.0)};
                                                           ^~~~~~~~~
                                                           static_cast<float>( )
      3 warnings and 4 errors generated.
      error: command '/opt/homebrew/opt/llvm/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: legacy-install-failure

× Encountered error while trying to install package.
╰─> GeodisTK
```

--- a/cpp/geodesic_distance_2d.cpp
+++ b/cpp/geodesic_distance_2d.cpp
@@ -9,7 +9,7 @@ using namespace std;
 float get_l2_distance(std::vector<float> p1, std::vector<float> p2)
 {
     float sq_sum = 0.0;
-    for(int d = 0; d < p1.size(); d++)
+    for(size_t d = 0; d < p1.size(); d++)
     {
         sq_sum = sq_sum + (p1[d] - p2[d]) * (p1[d] - p2[d]);
     }
@@ -20,7 +20,7 @@ float get_l2_distance(std::vector<float> p1, std::vector<float> p2)
 void insert_point_to_list(std::vector<Point2D> * list, int start_position,  Point2D p)
 {
     int insert_idx = list->size();
-    for(int i = start_position; i < list->size(); i++)
+    for(size_t i = start_position; i < list->size(); i++)
     {
         if(list->at(i).distance < p.distance)
         {
@@ -34,7 +34,7 @@ void insert_point_to_list(std::vector<Point2D> * list, int start_position,  Poin
 void update_point_in_list(std::vector<Point2D> * list, Point2D p)
 {
     int remove_idx = -1;
-    for(int i = 0; i < list->size(); i++)
+    for(size_t i = 0; i < list->size(); i++)
     {
         if(list->at(i).w == p.w && list->at(i).h == p.h)
         {
@@ -172,7 +172,7 @@ void geodesic2d_raster_scan(const float * img, const unsigned char * seeds, floa
         // forward scan
         int dh_f[4] = {-1, -1, -1, 0};
         int dw_f[4] = {-1, 0, 1, -1};
-        float local_dis_f[4] = {sqrt(2.0), 1.0, sqrt(2.0), 1.0};
+        float local_dis_f[4] = {sqrtf(2.0), 1.0, sqrtf(2.0), 1.0};
         // forward pass
         for(int h = 0; h < height; h++)
         {
@@ -200,7 +200,7 @@ void geodesic2d_raster_scan(const float * img, const unsigned char * seeds, floa
         // backward scann
         int dh_b[4] = {0, 1, 1, 1};
         int dw_b[4] = {1, -1, 0, 1};
-        float local_dis_b[4] = {1.0, sqrt(2.0), 1.0, sqrt(2.0)};
+        float local_dis_b[4] = {1.0, sqrtf(2.0), 1.0, sqrtf(2.0)};
         for(int h = height -1; h >= 0; h--)
         {
             for (int w = width - 1; w >= 0; w--)

